### ISSUE TITLE
Inline code style for admonition seems odd due to override

### ISSUE THREAD

**Issue by polarathene** [squidfunk/mkdocs-material#2492]
- [x] I've read the [contribution guidelines][1] and agree with them

__I've found a bug and checked that ...__

- [x] ... the problem doesn't occur with the default MkDocs template
- [x] ... the problem is not in any of my customizations (CSS, JS, template)
- [x] ... the documentation does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem

## Description

I don't quite get the reasoning for this override?:

https://github.com/squidfunk/mkdocs-material/blob/414ffe14cd880eabb234bd6d7ce884a62e0ca7d3/src/assets/stylesheets/main/extensions/markdown/_admonition.scss#L154-L163

How is the override visually better? If it was a concern for the user they might as well at that point just not use the `inline code syntax`?

If the concern is with contrast, wouldn't it be better to instead use alpha?

### Expected behavior

**EDIT:** See the comparison images below in the 2nd message.

<details>
<summary>original examples</summary>

Any of the following:

1. Restore bg-color:
   ![Screenshot_20210326_191218](https://user-images.githubusercontent.com/5098581/112590140-3496b600-8e67-11eb-83bc-3dbae7c6ff58.png)

2. Full restore:
   ![Screenshot_20210326_191806](https://user-images.githubusercontent.com/5098581/112590847-59d7f400-8e68-11eb-8257-94e2164109b4.png)

3. Change bg-color to brighter shade with opacity (`#ffffffbf`):
   ![Screenshot_20210326_192119](https://user-images.githubusercontent.com/5098581/112591040-9d326280-8e68-11eb-8dc7-dade7737aa63.png)

4. Change bg-color to darker shade with opacity (`#3a3a3a30`):
   ![Screenshot_20210326_193014](https://user-images.githubusercontent.com/5098581/112591776-c56e9100-8e69-11eb-80b2-acd5c680e71b.png)

5. Change bg-color to better match admonition colour (`#d7d7d7ad` with CSS [`mix-blend-mode: multiply`](https://developer.mozilla.org/en-US/docs/Web/CSS/mix-blend-mode)):
   ![Screenshot_20210326_193301](https://user-images.githubusercontent.com/5098581/112592112-4e85c800-8e6a-11eb-8575-6aba3eed57a8.png)

6. or with `mix-blend-mode: luminosity`:
   ![Screenshot_20210326_193734](https://user-images.githubusercontent.com/5098581/112592406-c9e77980-8e6a-11eb-9f20-d6848b9c0686.png)

Personally 3 or 6 look quite nice.

</details>

### Actual behavior

![Screenshot_20210326_191133](https://user-images.githubusercontent.com/5098581/112590096-2052b900-8e67-11eb-9e32-62fce2bb5d4a.png)

### Package versions / Project configuration / System information

Active release used for project demo site (if modified to use an inline code block in admonition title): https://squidfunk.github.io/mkdocs-material/reference/admonitions

**Comment by polarathene** [comment]
Replacing the referenced CSS rules to be only the 3 changes below with different blend modes looks much better personally. I've also included at the end how these look with current override, and override removed for comparison.

---

### Difference

```css
color: currentColor;
background-color: #bbbbbba3;
mix-blend-mode: difference;
```

![Screenshot_20210326_195545](https://user-images.githubusercontent.com/5098581/112594473-e0430480-8e6d-11eb-96d1-2d43a1201ae7.png)

---

### Color-Burn

```css
color: currentColor;
background-color: #444;
mix-blend-mode: color-burn;
```

![Screenshot_20210326_200928](https://user-images.githubusercontent.com/5098581/112595360-2fd60000-8e6f-11eb-8fd5-dae97fa1e9e0.png)

---

### Luminosity

```css
color: currentColor;
background-color: #ccc;
mix-blend-mode: luminosity;
```

![Screenshot_20210326_201339](https://user-images.githubusercontent.com/5098581/112595826-d6ba9c00-8e6f-11eb-8a08-6bb1900974d0.png)

---

### Multiply

```css
color: currentColor;
background-color: #e8e8e8;
mix-blend-mode: multiply;
```

![Screenshot_20210326_201804](https://user-images.githubusercontent.com/5098581/112596270-7841ed80-8e70-11eb-8eae-9a02f44fdd29.png)

---

### Just light bg with opacity:

```css
background-color: #ffffffb7
```

![Screenshot_20210326_202427](https://user-images.githubusercontent.com/5098581/112596942-5eed7100-8e71-11eb-83be-486580dd9588.png)

---

### All overrides removed:

![Screenshot_20210326_202527](https://user-images.githubusercontent.com/5098581/112597027-77f62200-8e71-11eb-866e-3151d47762d0.png)

---

### All overrides applied (current upstream style)

![Screenshot_20210326_202619](https://user-images.githubusercontent.com/5098581/112597135-9b20d180-8e71-11eb-93ff-6e0440e40cf3.png)

**Comment by squidfunk** [comment]
Thanks for your suggestions.

> If the concern is with contrast, wouldn't it be better to instead use alpha?

We've used alpha values for code block colors in the past, and it came with other problems, for example when using line numbers with the [`pymdownx-inline`](https://facelessuser.github.io/pymdown-extensions/extensions/highlight/#line-number-styles) method, which some users prefer to the table-based solution. The code was shining through the line numbers when scrolling, so we had to switch to opaque colors.

IIRC (it's a long time ago), we removed the background because we've used alpha values, so since that's no longer true, we might add it back. After all, it's more consistent, and I'm fine with the look. With the alpha approach, it looked really bad.
